File: 02_algorithm/programmers/Level2/이진 변환 반복하기.py
def solution(s):
    change_binary_count, eliminate_zero_count = 0, 0
    while True:
        change_binary_count += 1
        # 1. 모든 0 제거
        eliminate_zero_count += s.count('0')
        s = s.replace('0', '')
        # 2. c를 2진법으로 표현
        s_length = len(s)
        s = bin(s_length)[2:]
        if s == '1':
            break
    return [change_binary_count, eliminate_zero_count]


print(solution('110010101001'))
print(solution('01110'))
print(solution('1111111'))

File: 06_js/04_algorithm/programmers/Level2/이진 변환 반복하기.js
function solution(s) {
  let changeBinaryCount = 0;
  let eliminateZeroCount = 0;
  while (true) {
    changeBinaryCount += 1;
    const checkZero = s.match(/0/g);
    if (checkZero) {
      eliminateZeroCount += checkZero.length;
    }
    s = s.replace(/0/gi, '');
    s = s.length.toString(2);
    if (s === '1') {
      break
    }
  }
  return [changeBinaryCount, eliminateZeroCount];
}

console.log(solution('01110'));